flink-job/src/main/java/com/ververica/field/dynamicrules/functions/DynamicAlertFunction.java

```diff
@@ -91,8 +91,9 @@ public void processElement(
 
     Rule rule = ctx.getBroadcastState(Descriptors.rulesDescriptor).get(value.getId());
 
-    if (noRuleAvailable(rule)) {
-      log.error("Rule with ID {} does not exist", value.getId());
+    if (rule == null) {
```

Comment:
It seems that we should instead add a mitigation strategy for this occurrence instead. As I expected, this question get's raised.
http://disq.us/p/288ot93  (question from Theo D)

Reply:
I would object - we're not building a product here and can leave a few things open. What you should do here, though, may even depend on the concrete use case.
I would agree to leave a `TODO` item here, if you wish.

Reply:
Let's go for a comment.